framework/audio/internal/audiostream.h

```diff
@@ -0,0 +1,61 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_AUDIOSTREAM_H
+#define MU_AUDIO_AUDIOSTREAM_H
+
+#include <vector>
+#include "audio/iaudiostream.h"
+#include "src.h"
+
+namespace mu {
+namespace audio {
+class AudioStream : public IAudioStream
+{
+public:
+    AudioStream();
+
+    //! load data from file wav, mp3 or ogg (automatically checked)
+    bool loadFile(const mu::io::path& path) override;
+
+    bool loadMP3FromMemory(const void* pData, size_t dataSize);
```

Comment:
Seems that you can unite loadMP3FromMemory and loadFile by replacing the path with IODevice